tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,891 @@
+import blaze as bz
+import itertools
+from nose.tools import assert_true
+from nose_parameterized import parameterized
+import numpy as np
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+from toolz import merge
+
+from zipline.pipeline import SimplePipelineEngine, Pipeline, CustomFactor
+from zipline.pipeline.common import (
+    EVENT_DATE_FIELD_NAME,
+    FISCAL_QUARTER_FIELD_NAME,
+    FISCAL_YEAR_FIELD_NAME,
+    SID_FIELD_NAME,
+    TS_FIELD_NAME,
+)
+from zipline.pipeline.data import DataSet
+from zipline.pipeline.data import Column
+from zipline.pipeline.loaders.blaze.estimates import (
+    BlazeNextEstimatesLoader,
+    BlazePreviousEstimatesLoader
+)
+from zipline.pipeline.loaders.quarter_estimates import (
+    NextQuartersEstimatesLoader,
+    normalize_quarters,
+    PreviousQuartersEstimatesLoader,
+    split_normalized_quarters,
+)
+from zipline.testing.fixtures import (
+    WithAssetFinder,
+    WithTradingSessions,
+    ZiplineTestCase,
+)
+from zipline.testing.predicates import assert_equal
+from zipline.utils.numpy_utils import datetime64ns_dtype
+from zipline.utils.numpy_utils import float64_dtype
+
+
+class Estimates(DataSet):
+    event_date = Column(dtype=datetime64ns_dtype)
+    fiscal_quarter = Column(dtype=float64_dtype)
+    fiscal_year = Column(dtype=float64_dtype)
+    estimate = Column(dtype=float64_dtype)
+
+
+def QuartersEstimates(num_qtr):
+    class QtrEstimates(Estimates):
+        num_quarters = num_qtr
+        name = Estimates
+    return QtrEstimates
+
+
+class WithEstimates(WithTradingSessions, WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing cls.loader and cls.events as class
+    level fixtures.
+
+
+    Methods
+    -------
+    make_loader() -> PipelineLoader
+        Method which returns the loader to be used throughout tests.
+    """
+
+    # Short window defined in order for test to run faster.
+    START_DATE = pd.Timestamp('2014-12-28')
+    END_DATE = pd.Timestamp('2015-02-03')
+
+    @classmethod
+    def make_loader(cls, events, columns):
+        return None
```

Comment:
maybe change this to: `raise NotImplementedError('make_loader')` so users know they must implement this.